lua/class_definitions.lua

```diff
@@ -75,6 +75,24 @@ classes = {
         functions = {
         }
     },
+    calendar = {
+        attributes = {
+        },
+        functions = {
+            { name = "get_turn", rval = "int", args = { } },
+            { name = "minutes_past_midnight", rval = "int", args = { } },
+            { name = "seconds_past_midnight", rval = "int", args = { } },
+            { name = "sunrise", rval = "calendar", args = { } },
+            { name = "sunset", rval = "calendar", args = { } },
+            { name = "is_night", rval = "bool", args = { } },
+            { name = "sunlight", rval = "calendar", args = { } },
```

Comment:
From the top of my head, you need to
- add some markers in lua/classbindings.lua to distinguish function that return by value and those that return by reference.
- add code in generate_binding.lua to generate the proper wrappers.
- add code in catalua.cpp (probably to the existing LuaValue/LuaReference or make a new type or something else) to support this. That's probably the most complicated part.